pkg/services/annotations/annotationsimpl/xorm_store.go

```diff
@@ -64,9 +64,63 @@ func (r *xormRepositoryImpl) Add(ctx context.Context, item *annotations.Item) er
 		if _, err := sess.Table("annotation").Insert(item); err != nil {
 			return err
 		}
+		return r.synchronizeTags(ctx, item)
+	})
+}
+
+// AddMany inserts large batches of annotations at once.
+// It does not return IDs associated with created annotations, and it does not support annotations with tags. If you need this functionality, use the single-item Add instead.
+// This is due to a limitation with some supported databases:
+// We cannot correlate the IDs of batch-inserted records without acquiring a full table lock in MySQL.
+// Annotations have no other uniquifier field, so we also cannot re-query for them after the fact.
+// So, callers can only reliably use this endpoint if they don't care about returned IDs.
+func (r *xormRepositoryImpl) AddMany(ctx context.Context, items []annotations.Item) error {
```

Comment:
You asked before about the `alertingBigTransactions` feature toggle. There's no reason we can't keep the toggle, if it saves you a little overhead, or we could define a similar toggle `annotationsBigTransaction` and use it, but I think we need to have a control in place to switch big batches on and off. If it's switched off, you can just fall down to the code you use to save items with tags.